src/Coin.cpp

```diff
@@ -278,6 +278,14 @@ Hash::Hasher TW::base58Hasher(TWCoinType coin) {
     return getCoinInfo(coin).base58Hasher;
 }
 
+uint32_t TW::slip44Id(TWCoinType coin) {
+    auto id = getCoinInfo(coin).numericId;
+    switch (coin) {
```

Comment:
This smells hacking here!  Why not have a separate slip44Id in coins.json?  Too much redundancy?

Reply:
If we use a hidden exceptional logic here, we could just as well use a custom derivation path for BSC (m/44/714000/0...), and turn it back to 714 with a similar switch somewhere, and then get away without any new parameters and breaking change.  Not good idea.

Reply:
I know this code is ugly, I feel like `CoinInfoData` is not flexible to use, ideally I'd like to access it like json or map: check if it has extra `slip44` then returns it.

Reply:
we need to decouple coin type from derivation path, so the latter you said is even worse

Reply:
`CoinInfoData` is just a C++ struct represetation of `coins.json`.  Optional slip44Id is possible in coins.json, and handled in generator script: if present use that, if not, use numerical Id, to produce CoinInfoData.slip44Id.

Reply:
I see you did exactly that :)